tfhe/src/integer/key_switching_key/test.rs

```diff
@@ -75,13 +80,17 @@ fn gen_multi_keys_test_crtinteger_to_crtinteger_ci_run_filter() {
 fn gen_multi_keys_test_crtinteger_to_crtinteger_fail_ci_run_filter() {
     let basis = vec![2, 3, 5, 7, 11];
 
-    // We generate a set of client/server keys, using the default parameters:
+    // We retrieve only one keys set from cache since testing key switching with two identical keys
+    // set is meaningless.
     let (client_key_1, server_key_1) =
-        crate::integer::gen_keys_crt(PARAM_MESSAGE_2_CARRY_2_KS_PBS, basis.clone());
+        KEY_CACHE.get_from_params(PARAM_MESSAGE_2_CARRY_2_KS_PBS, IntegerKeyKind::CRT);
+    let client_key_1 = CrtClientKey::from((client_key_1, basis.clone()));
 
-    // We generate a set of client/server keys, using the default parameters:
+    // We retrieve only one keys set from cache since testing key switching with two identical keys
+    // set is meaningless.
```

Comment:
I think the comment is slightly wrong we retrieve both from the keycache but they are from different parameter sets